### ISSUE TITLE
supporting flexibly retaining study estimates in meta-analysis output dataframe

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/parameters#348]
It would be nice to have an argument in `model_parameters`, similar to `include_studies` argument for `broom::tidy.rma`, to give the user the ability to retain or drop the study estimates from the final dataframe.

**Comment by IndrajeetPatil** [comment]
This will be relevant for all meta-analysis objects, like `rma`, `meta_random`, `metaplus`, etc.

**Comment by strengejacke** [comment]
``` r
library(parameters)
library(metafor)
#> Loading required package: Matrix
#> Loading 'metafor' package (version 2.4-0). For an overview 
#> and introduction to the package please type: help(metafor).
library(metaBMA)
#> Loading required package: Rcpp
library(metaplus)
#> 
#> Note: The second parameter is the study standard error not the standard error squared as in the metafor package.

mydat <- data.frame(
  effectsize = c(-0.393, 0.675, 0.282, -1.398),
  stderr = c(0.317, 0.317, 0.13, 0.36)
)
model <- rma(yi = effectsize, sei = stderr, method = "REML", data = mydat)

model_parameters(model)
#> Parameter | Coefficient |   SE |         95% CI |     z |      p | Weight
#> -------------------------------------------------------------------------
#> Study 1   |       -0.39 | 0.32 | [-1.01,  0.23] | -1.24 | 0.215  |   9.95
#> Study 2   |        0.68 | 0.32 | [ 0.05,  1.30] |  2.13 | 0.033  |   9.95
#> Study 3   |        0.28 | 0.13 | [ 0.03,  0.54] |  2.17 | 0.030  |  59.17
#> Study 4   |       -1.40 | 0.36 | [-2.10, -0.69] | -3.88 | < .001 |   7.72
#> Overall   |       -0.18 | 0.44 | [-1.05,  0.68] | -0.42 | 0.676  |
model_parameters(model, include_studies = FALSE)
#> Parameter | Coefficient |   SE |        95% CI |     z |     p
#> --------------------------------------------------------------
#> Overall   |       -0.18 | 0.44 | [-1.05, 0.68] | -0.42 | 0.676

data(dat.bcg)
dat <- escalc(
  measure = "RR",
  ai = tpos,
  bi = tneg,
  ci = cpos,
  di = cneg,
  data = dat.bcg
)
dat$alloc <- ifelse(dat$alloc == "random", "random", "other")
model <- rma(yi, vi, mods = ~alloc, data = dat, digits = 3, slab = author)

model_parameters(model)
#> # Random Effects
#> 
#> Parameter         | Coefficient |   SE |         95% CI |      z |      p | Weight
#> ----------------------------------------------------------------------------------
#> Aronson           |       -0.89 | 0.57 | [-2.01,  0.23] |  -1.56 | 0.119  |   3.07
#> Ferguson & Simes  |       -1.59 | 0.44 | [-2.45, -0.72] |  -3.59 | < .001 |   5.14
#> Rosenthal et al.1 |       -1.35 | 0.64 | [-2.61, -0.08] |  -2.09 | 0.036  |   2.41
#> Hart & Sutherland |       -1.44 | 0.14 | [-1.72, -1.16] | -10.19 | < .001 |  49.97
#> Vandiviere et al  |       -1.62 | 0.47 | [-2.55, -0.70] |  -3.43 | < .001 |   4.48
#> TPT Madras        |        0.01 | 0.06 | [-0.11,  0.14] |   0.19 | 0.849  | 252.42
#> Coetzee & Berjak  |       -0.47 | 0.24 | [-0.94,  0.00] |  -1.98 | 0.048  |  17.72
#> Overall           |       -0.49 | 0.36 | [-1.20,  0.22] |  -1.35 | 0.176  |       
#> 
#> # other
#> 
#> Parameter            | Coefficient |   SE |         95% CI |     z |      p | Weight
#> ------------------------------------------------------------------------------------
#> Frimodt-Moller et al |       -0.22 | 0.23 | [-0.66,  0.23] | -0.96 | 0.336  |  19.53
#> Stein & Aronson      |       -0.79 | 0.08 | [-0.95, -0.62] | -9.46 | < .001 | 144.81
#> Rosenthal et al.2    |       -1.37 | 0.27 | [-1.90, -0.84] | -5.07 | < .001 |  13.69
#> Comstock et al.1     |       -0.34 | 0.11 | [-0.56, -0.12] | -3.05 | 0.002  |  80.57
#> Comstock & Webster   |        0.45 | 0.73 | [-0.98,  1.88] |  0.61 | 0.541  |   1.88
#> Comstock et al.2     |       -0.02 | 0.27 | [-0.54,  0.51] | -0.06 | 0.948  |  14.00
#> Overall              |       -0.47 | 0.26 | [-0.97,  0.04] | -1.82 | 0.069  |
model_parameters(model, include_studies = FALSE)
#> # other
#> 
#> Parameter | Coefficient |   SE |        95% CI |     z |     p
#> --------------------------------------------------------------
#> Overall   |       -0.47 | 0.26 | [-0.97, 0.04] | -1.82 | 0.069
#> 
#> # Random Effects
#> 
#> Parameter | Coefficient |   SE |        95% CI |     z |     p
#> --------------------------------------------------------------
#> Overall   |       -0.49 | 0.36 | [-1.20, 0.22] | -1.35 | 0.176

data(towels)
set.seed(123)
mb <- meta_bma(logOR, SE, study, towels,
               d = prior("norm", c(mean=0, sd=.3), lower=0),
               tau = prior("invgamma", c(shape = 1, scale = 0.15)),
               rel.tol = .Machine$double.eps^.15, iter=1000)

model_parameters(mb)
#> # Studies
#> 
#> Parameter                                           | Coefficient |   SE |        95% CI | Weight
#> -------------------------------------------------------------------------------------------------
#> Goldstein, Cialdini, & Griskevicius (2008), Exp. 1  |        0.38 | 0.20 | [-0.01, 0.77] |  25.59
#> Goldstein, Cialdini, & Griskevicius  (2008), Exp. 2 |        0.30 | 0.14 | [ 0.04, 0.57] |  53.97
#> Schultz, Khazian, & Zaleski (2008), Exp. 2          |        0.21 | 0.19 | [-0.17, 0.58] |  27.24
#> Schultz, Khazian, & Zaleski (2008), Exp. 3          |        0.25 | 0.17 | [-0.08, 0.58] |  34.57
#> Mair & Bergin-Seers (2010), Exp. 1                  |        0.29 | 0.82 | [-1.33, 1.90] |   1.47
#> Bohner & Schluter (2014), Exp. 1                    |       -0.12 | 0.25 | [-0.61, 0.36] |  16.25
#> Bohner & Schluter (2014), Exp. 2                    |       -1.46 | 0.76 | [-2.95, 0.03] |   1.73
#> 
#> # Meta-Parameters
#> 
#> Parameter | Coefficient |   SE |       95% CI |    BF |  Rhat |    ESS
#> ----------------------------------------------------------------------
#> averaged  |        0.21 | 0.08 | [0.04, 0.36] |       |       |       
#> fixed     |        0.21 | 0.07 | [0.08, 0.37] | 23.87 | 1.006 | 449.50
#> random    |        0.19 | 0.09 | [0.01, 0.35] | 10.26 | 1.001 | 790.10
#> 
#> Using highest density intervals as credible intervals.
model_parameters(mb, include_studies = FALSE)
#> Parameter | Coefficient |   SE |       95% CI |    BF |  Rhat |    ESS
#> ----------------------------------------------------------------------
#> averaged  |        0.21 | 0.08 | [0.04, 0.36] |       |       |       
#> fixed     |        0.21 | 0.07 | [0.08, 0.37] | 23.87 | 1.006 | 449.50
#> random    |        0.19 | 0.09 | [0.01, 0.35] | 10.26 | 1.001 | 790.10
#> 
#> Using highest density intervals as credible intervals.


mf <- meta_fixed(logOR, SE, study, data = towels,
                 d = prior("norm", c(mean=0, sd=.3), lower=0))

model_parameters(mf)
#> # Studies
#> 
#> Parameter                                           | Coefficient |   SE |        95% CI | Weight
#> -------------------------------------------------------------------------------------------------
#> Goldstein, Cialdini, & Griskevicius (2008), Exp. 1  |        0.38 | 0.20 | [-0.01, 0.77] |  25.59
#> Goldstein, Cialdini, & Griskevicius  (2008), Exp. 2 |        0.30 | 0.14 | [ 0.04, 0.57] |  53.97
#> Schultz, Khazian, & Zaleski (2008), Exp. 2          |        0.21 | 0.19 | [-0.17, 0.58] |  27.24
#> Schultz, Khazian, & Zaleski (2008), Exp. 3          |        0.25 | 0.17 | [-0.08, 0.58] |  34.57
#> Mair & Bergin-Seers (2010), Exp. 1                  |        0.29 | 0.82 | [-1.33, 1.90] |   1.47
#> Bohner & Schluter (2014), Exp. 1                    |       -0.12 | 0.25 | [-0.61, 0.36] |  16.25
#> Bohner & Schluter (2014), Exp. 2                    |       -1.46 | 0.76 | [-2.95, 0.03] |   1.73
#> 
#> # Meta-Parameters
#> 
#> Parameter | Coefficient |   SE |       95% CI |    BF |            Prior
#> ------------------------------------------------------------------------
#> Overall   |        0.21 | 0.08 | [0.06, 0.36] | 23.87 | Norm (0 +- 0.30)
#> 
#> Using highest density intervals as credible intervals.
model_parameters(mf, include_studies = FALSE)
#> Parameter | Coefficient |   SE |       95% CI |    BF |            Prior
#> ------------------------------------------------------------------------
#> Overall   |        0.21 | 0.08 | [0.06, 0.36] | 23.87 | Norm (0 +- 0.30)
#> 
#> Using highest density intervals as credible intervals.

mr <- meta_random(logOR, SE, study, data = towels,
                  d = prior("norm", c(mean=0, sd=.3), lower = 0),
                  tau = prior("invgamma", c(shape = 1, scale = 0.15)),
                  rel.tol=.Machine$double.eps^.15, iter=1000)

model_parameters(mr)
#> # Studies
#> 
#> Parameter                                           | Coefficient |   SE |        95% CI | Weight
#> -------------------------------------------------------------------------------------------------
#> Goldstein, Cialdini, & Griskevicius (2008), Exp. 1  |        0.38 | 0.20 | [-0.01, 0.77] |  25.59
#> Goldstein, Cialdini, & Griskevicius  (2008), Exp. 2 |        0.30 | 0.14 | [ 0.04, 0.57] |  53.97
#> Schultz, Khazian, & Zaleski (2008), Exp. 2          |        0.21 | 0.19 | [-0.17, 0.58] |  27.24
#> Schultz, Khazian, & Zaleski (2008), Exp. 3          |        0.25 | 0.17 | [-0.08, 0.58] |  34.57
#> Mair & Bergin-Seers (2010), Exp. 1                  |        0.29 | 0.82 | [-1.33, 1.90] |   1.47
#> Bohner & Schluter (2014), Exp. 1                    |       -0.12 | 0.25 | [-0.61, 0.36] |  16.25
#> Bohner & Schluter (2014), Exp. 2                    |       -1.46 | 0.76 | [-2.95, 0.03] |   1.73
#> 
#> # Meta-Parameters
#> 
#> Parameter | Coefficient |   SE |       95% CI |   BF |  Rhat |     ESS |                Prior
#> ---------------------------------------------------------------------------------------------
#> Overall   |        0.19 | 0.09 | [0.02, 0.35] | 3.81 | 1.001 | 1197.70 |     Norm (0 +- 0.30)
#> tau       |        0.13 | 0.09 | [0.02, 0.30] |      | 1.005 | 1051.70 | Invgamma (1 +- 0.15)
#> 
#> Using highest density intervals as credible intervals.
model_parameters(mr, include_studies = FALSE)
#> Parameter | Coefficient |   SE |       95% CI |   BF |  Rhat |     ESS |                Prior
#> ---------------------------------------------------------------------------------------------
#> Overall   |        0.19 | 0.09 | [0.02, 0.35] | 3.81 | 1.001 | 1197.70 |     Norm (0 +- 0.30)
#> tau       |        0.13 | 0.09 | [0.02, 0.30] |      | 1.005 | 1051.70 | Invgamma (1 +- 0.15)
#> 
#> Using highest density intervals as credible intervals.


mp <- metaplus(yi, sei, slab = study, data = mag)
model_parameters(mp)
#> Parameter    | Coefficient |   SE |         95% CI |     z |      p | Weight
#> ----------------------------------------------------------------------------
#> Morton       |       -0.83 | 1.25 | [-3.27,  1.61] | -0.67 | 0.505  |   0.80
#> Rasmussen    |       -1.06 | 0.41 | [-1.87, -0.24] | -2.55 | 0.011  |   2.42
#> Smith        |       -1.28 | 0.81 | [-2.86,  0.31] | -1.58 | 0.114  |   1.24
#> Abraham      |       -0.04 | 1.43 | [-2.85,  2.76] | -0.03 | 0.976  |   0.70
#> Feldstedt    |        0.22 | 0.49 | [-0.74,  1.18] |  0.46 | 0.648  |   2.04
#> Schechter    |       -2.41 | 1.07 | [-4.51, -0.31] | -2.25 | 0.025  |   0.93
#> Ceremuzynski |       -1.28 | 1.19 | [-3.62,  1.06] | -1.07 | 0.283  |   0.84
#> Bertschat    |       -1.19 | 1.66 | [-4.45,  2.06] | -0.72 | 0.473  |   0.60
#> Singh        |       -0.70 | 0.54 | [-1.75,  0.36] | -1.30 | 0.194  |   1.87
#> Pereira      |       -2.21 | 1.11 | [-4.38, -0.03] | -1.99 | 0.047  |   0.90
#> Schechter 1  |       -2.04 | 0.78 | [-3.57, -0.51] | -2.61 | 0.009  |   1.28
#> Golf         |       -0.85 | 0.62 | [-2.06,  0.36] | -1.37 | 0.169  |   1.62
#> Thogersen    |       -0.79 | 0.63 | [-2.02,  0.43] | -1.27 | 0.205  |   1.60
#> LIMIT-2      |       -0.30 | 0.15 | [-0.59, -0.01] | -2.04 | 0.041  |   6.82
#> Schechter 2  |       -1.57 | 0.57 | [-2.70, -0.45] | -2.74 | 0.006  |   1.74
#> ISIS-4       |        0.06 | 0.03 | [ 0.00,  0.12] |  1.82 | 0.069  |  31.60
#> Overall      |       -0.75 | 0.23 | [-1.26, -0.34] | -3.20 | < .001 |
model_parameters(mp, include_studies = FALSE)
#> Parameter | Coefficient |   SE |         95% CI |     z |      p
#> ----------------------------------------------------------------
#> Overall   |       -0.75 | 0.23 | [-1.26, -0.34] | -3.20 | < .001
```

<sup>Created on 2020-11-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by IndrajeetPatil** [comment]
Awesome, thanks!!
